### PR TITLE
MAINT: sparse.linalg: Change ambiguous parameter name `tol` in sparse linear solver into `rtol`

### PR DESCRIPTION
**Pull request by zhaog6**
#### What does this implement/fix?

In some sparse linear iterative solvers, the original parameter name `tol` is ambiguous, this PR will replace `tol` by `rtol` to represent relative residual tolerance. Users will be more clear about the meaning of `rtol`. In addition, refer also to the discussion on `rtol` in the following link https://github.com/scipy/scipy/pull/16202#discussion_r884367708.

### CODE REVIEW

File: scipy/sparse/linalg/_isolve/_gcrotmk.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ def _fgmres(matvec, v0, m, atol, lpsolve=None, rpsolve=None, cs=(), outer_v=(),
     return Q, R, B, vs, zs, y, res
 
 
-def gcrotmk(A, b, x0=None, tol=1e-5, maxiter=1000, M=None, callback=None,
+def gcrotmk(A, b, x0=None, rtol=1e-5, maxiter=1000, M=None, callback=None,
```
Comment (tylerjereddy) [inline scipy/sparse/linalg/_isolve/_gcrotmk.py:182]: Sorry if I missed it in the linked discussion--did the team say they were "ok" with breaking code that currently uses the `tol` keyword argument without a warning/deprecation period? I think this and a few other public functions may be involved in the transition, so just want to make sure this was discussed.

Comment (zhaog6) [reply scipy/sparse/linalg/_isolve/_gcrotmk.py:182]: > did the team say they were "ok" with breaking code that currently uses the tol keyword argument without a warning/deprecation period?

Thank you @tylerjereddy for your review. It has not been discussed in detailed. Only SYMMLQ function (a new Krylov method) has been discussed in PR #16202. @ilayn

**Comment by mdhaber** [comment]
@zhaog6 I don't think we would want to break backward compatibility like this, but you can use the [`_rename_parameter` decorator](https://github.com/scipy/scipy/blob/773ee7611deb9390bbc7d0e0a8d449b1f6d63981/scipy/_lib/_util.py#L650) to accept both the old and new names to maintain backward compatibility. Would you like to do that?

**Comment by mdhaber** [comment]
Please do re-open this if you'd like to do this in a backward compatible way (https://github.com/scipy/scipy/pull/16480#issuecomment-1261847773), but for now I'll close because I haven't heard back. Thanks @zhaog6.

**Comment by zhaog6** [comment]
@mdhaber , sorry, I haven't seen this for too long. Okay, no problem.

**Pull request closed without merging**